src/Compilers/Core/Portable/PEWriter/Types.cs

```diff
@@ -397,23 +397,23 @@ internal interface IManagedPointerTypeReference : ITypeReference
         ITypeReference GetTargetType(EmitContext context);
     }
 
-    internal struct InterfaceImplementation
+    internal struct TypeReferenceWithAttributes
     {
         /// <summary>
-        /// The interface being implemented.
+        /// The type reference.
         /// </summary>
-        public ITypeReference Interface { get; }
+        public ITypeReference TypeRef { get; }
 
         /// <summary>
-        /// The attributes on the interface implementation itself.
+        /// The attributes on the type reference itself.
         /// </summary>
         public ImmutableArray<ICustomAttribute> Attributes { get; }
 
-        public InterfaceImplementation(
-            ITypeReference iface,
+        public TypeReferenceWithAttributes(
+            ITypeReference typeRef,
             ImmutableArray<ICustomAttribute> attributes)
         {
-            Interface = iface;
+            TypeRef = typeRef;
             Attributes = attributes;
```

Comment:
Or, alternatively, make `attributes` an optional parameter and set `Attributes` to `Empty` in that case. That might simplify a few callers.